### PR TITLE
Add Broadcom SOTP PTA

### PR DESCRIPTION
**Pull request by shitalt**
Add Broadcom SOTP PTA

Signed-off-by: Rahul Gupta <rahul.gupta@broadcom.com>
Signed-off-by: Sheetal Tigadoli <sheetal.tigadoli@broadcom.com>

### CODE REVIEW

File: core/pta/bcm/sotp.c [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+	return res;
+}
+
+pseudo_ta_register(.uuid = SOTP_SERVICE_UUID,
+		   .name = SOTP_TA_NAME,
+		   .flags = PTA_DEFAULT_FLAGS,
+		   .create_entry_point = create_ta,
+		   .destroy_entry_point = destroy_ta,
+		   .open_session_entry_point = open_session,
+		   .close_session_entry_point = close_session,
```
Comment (jforissier) [inline core/pta/bcm/sotp.c]: You can drop these since they're basically useless.

Comment (shitalt) [reply core/pta/bcm/sotp.c]: Will remove these

**Comment by shitalt** [comment]
When rebasing, will update the comment description as shown below 

`Add Broadcom SOTP PTA.`
`This PTA is used by userspace application to read`
`SOTP value and determine modules/features that are`
`enabled on this platform`

**Comment by jforissier** [comment]
`Reviewed-by: Jerome Forissier <jerome@forissier.org>`

**Comment by shitalt** [comment]
Rebased and Tags applied

**Pull request merged**